packages/server-core/src/projects/project/project-helper.ts

```diff
@@ -778,63 +782,87 @@ export const findBuilderTags = async (): Promise<Array<ProjectBuilderTagsType>>
   const publicECRExec = publicECRRepoRegex.exec(builderRepo)
   const privateECRExec = privateECRRepoRegex.exec(builderRepo)
   if (publicECRExec) {
+    const awsCredentials = `[default]\naws_access_key_id=${config.aws.eks.accessKeyId}\naws_secret_access_key=${config.aws.eks.secretAccessKey}\n[role]\nrole_arn = ${config.aws.eks.roleArn}\nsource_profile = default`
+
+    if (!fs.existsSync(awsPath)) fs.mkdirSync(awsPath, { recursive: true })
+    fs.writeFileSync(credentialsPath, Buffer.from(awsCredentials))
+
     const ecr = new ECRPUBLICClient({
-      credentials: {
-        accessKeyId: config.aws.eks.accessKeyId,
-        secretAccessKey: config.aws.eks.secretAccessKey
-      },
+      credentials: fromIni({
+        profile: config.aws.eks.roleArn ? 'role' : 'default',
+        filepath: credentialsPath
+      }),
```

Comment:
Why are we creating a cred file each time, instead of just passing the values to client when we instantiate it?

Reply:
I updated it to not write the credentials file if it exists.

Reply:
I was more asking about why we are using cred files, instead of passing those values to the client directly. Is it a paradigm across the app to write cred files, or will other parts of the applications expect credential files?

Reply:
When I was looking at authenticating the AWS SDK with Role ARNs, it seemed like the only way to do that was from a credentials file: https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/setting-credentials-node.html

Reply:
You can use the assume_role command on the client. But it does seem like the syntax is a bit annoying for the JS SDK: https://docs.aws.amazon.com/code-library/latest/ug/sts_example_sts_AssumeRole_section.html

Since this is an admin only thing, probably not a huge deal to have this extra file I/O.